values/thanos/thanos.gotmpl

```diff
@@ -0,0 +1,113 @@
+{{- $v := .Values }}
+{{- $vl:= $v.apps.thanos }}
+{{- $obj := $vl.objstore }}
+{{- $sp := $obj.storageProvider | get "type" dict }}
+
+## @section Thanos Query parameters
+query:
+  enabled: true
+  replicaCount: {{ $vl.query.replicaCount }}
+  dnsDiscovery:
+    enabled: false
+  stores:
+    - "thanos-storegateway:10901"
+    {{- range $i := until $vl.receiver.replicaCount }}
+    - "thanos-receive-{{ $i }}.thanos-receive-headless.monitoring.svc.cluster.local:10901"
+    {{- end }}
+  resources: {{- $vl.query.resources | toYaml | nindent 4 }}
+  extraFlags:
+    - "--query.auto-downsampling"
+
+## @section Thanos Query Frontend parameters
+queryFrontend:
+  enabled: true
+  resources: {{- $vl.queryFrontend.resources | toYaml | nindent 4 }}
+
+## @section Thanos Bucket Web parameters
+bucketweb:
+  enabled: true
+  resources: {{- $vl.bucketweb.resources | toYaml | nindent 4 }}
+
+## @section Thanos Compactor parameters
+compactor:
+  enabled: true
+  ## By default, there is NO retention set for object storage data. This means that data is stored forever, 
+  ## which is a valid and recommended way of running Thanos. Defauls are therefor set to 0s
+  retentionResolutionRaw: {{ $vl.compactor.retentionResolutionRaw }}
+  retentionResolution5m: {{ $vl.compactor.retentionResolution5m }}
+  retentionResolution1h: {{ $vl.compactor.retentionResolution1h }}
+  persistence:
+    size: {{ $vl.compactor.persistence.size }}
+  resources: {{- $vl.compactor.resources | toYaml | nindent 4 }}
+
+## @section Thanos Store Gateway parameters
+storegateway:
+  enabled: true
+  persistence:
+    size: {{ $vl.storegateway.persistence.size }}
+  resources: {{- $vl.storegateway.resources | toYaml | nindent 4 }}
+
+## @section Thanos Receive parameters
+receive:
+  enabled: true
+  mode: {{ $vl.receiver.mode }}
+  replicaCount: {{ $vl.receiver.replicaCount }}
+  persistence:
+    enabled: true
+    size: {{ $vl.receiver.persistence.size }}
+  resources: {{- $vl.receiver.resources | toYaml | nindent 4 }}
+  ## How long to retain raw samples on local storage. 0d - disables this retention. Default is set to 7d
+  tsdbRetention: 7d
```

Comment:
I am not sure if grouping horizontally (resources/sizes) instead of vertically (per sub-app) is a good approach. 
Regard the retention it is strongly related to the disk size - a user should see in the same context size and retention IMO

Reply:
For resources and persistence, I agree on grouping, but I don't agree on retention because it is very app specific.